### DISCUSSION TITLE
STM32F429I-DISC1 Support

### DISCUSSION THREAD

**Discussion by jberry** [shanteacontrols/OpenDeck#250] [Q&A] [answered]
I am looking for a US-available board that could support 56 buttons + 5din MIDI. 
Given all of the I/O breakout on the STM32F429I-DISC1 board, is 56 button inputs possible with a custom YAML? Is there anything that makes a pin NOT work for a button input?

**Comment by paradajz** [comment, marked as answer]
>STM32F429I-DISC1

The MCU on that board currently isn't supported, but it's more or less trivial to add support for it.

> is 56 button inputs possible with a custom YAML

Yes.

> Is there anything that makes a pin NOT work for a button input?

Depends on the MCU itself and the development board you're using. On STM32F4 series, pretty much any pin can be used as an input. However, if a certain pin is hardwired on a peripheral of some sort on that development board, that would make it hard to use as either output or input, depending of course on what it is wired to.

**Comment by jberry** [comment]
Thanks for the quick reply! 
This other devkit looks to use an almost identical chip to the Waveshare 407 series, https://a.co/d/dvfGpH2; 
Is there somewhere specific I can find the currently supported chips? This devkit without the touchscreen/less goodies looks like it would have a better chance at not having peripheral conflicts.

**Reply by paradajz** [reply]
That specific board is explicitly supported, I actually had it on my desk a while ago:
https://github.com/shanteacontrols/OpenDeck/blob/master/config/target/stm32f4ve.yml

> Is there somewhere specific I can find the currently supported chips?

Check here:
https://github.com/shanteacontrols/OpenDeck/tree/master/config/mcu

**Comment by jberry** [comment]
Is there anything you remember that would prevent me from using the Analog & LED pins as button inputs instead via a custom YAML? That'd get me to 58 buttons if I stole those pins

**Comment by paradajz** [comment]
Check the schematic
https://stm32-base.org/boards/STM32F407VET6-STM32-F4VE-V2.0.html

Some pins have LEDs connected to them, but otherwise you're good.
